lib/HttpClient.js

```diff
@@ -788,6 +793,9 @@ HttpClient.prototype.request = function request(opts, cb) {
     assert.object(opts, 'options');
     assert.func(cb, 'callback');
 
+    var self = this;
+    self._inflightRequests++;
```

Comment:
It seems that this increments the inflight requests number only for the initial request, but not for retries? On the other hand it seems we decrement number of inflight requests on raw requests' events? If that's correct it seems we'd want that behavior to be symmetrical, otherwise we'd run into cases where the number of inflight requests is < 0.

Reply:
Great catch! I thought I would catch this with the connection timeout test but it didn't because I set `retries: 1` 🤦‍♂️